### PR TITLE
Fix remove outdated item by allocator

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: spaceallocator/spaceallocator.go [hunk 1]
```diff
@@ -331,14 +340,59 @@ func (allocator *Allocator) allocateDone() error {
...
+
+		log.WithFields(log.Fields{
+			"mountPoint": allocator.part.mountPoint, "id": item.id, "size": item.size,
+		}).Debug("Remove outdated item")
+
+		if err := item.allocator.remover(item.id); err != nil {
+			return freedSize, err
+		}
+
+		item.allocator.part.freeSpace(item.size)
```
Comment (MykolaSolyanko) [inline spaceallocator/spaceallocator.go:395]: delete this because below you `freeSpace` for the freedSize

**Review by MykolaSolyanko — APPROVED** [review]

File: spaceallocator/spaceallocator.go [hunk 2]
```diff
@@ -175,11 +188,11 @@ func (allocator *Allocator) AllocateSpace(size uint64) (*Space, error) {
 		return nil, err
 	}
 
-	return &Space{size: size, path: allocator.path, allocator: allocator}, nil
+	return &spaceInstance{size: size, path: allocator.path, allocator: allocator}, nil
 }
 
 // Accept accepts previously allocated space.
-func (space *Space) Accept() error {
+func (space *spaceInstance) Accept() (err error) {
```
Comment (Eabramov84) [inline spaceallocator/spaceallocator.go:195]: why you chaged to named return in this case?

File: spaceallocator/spaceallocator.go [hunk 3]
```diff
@@ -190,11 +203,11 @@ func (space *Space) Accept() error {
 		return err
 	}
 
-	return nil
+	return err
 }
 
 // Release releases previously allocated space.
-func (space *Space) Release() error {
+func (space *spaceInstance) Release() (err error) {
```
Comment (Eabramov84) [inline spaceallocator/spaceallocator.go:210]: the same

File: spaceallocator/spaceallocator.go [hunk 4]
```diff
@@ -474,10 +528,12 @@ func (part *partition) removeOutdatedItems(requiredSize uint64) (freedSize uint6
 			"mountPoint": part.mountPoint, "id": item.id, "size": item.size,
 		}).Debug("Remove outdated item")
 
-		if err = item.allocator.removeOutdatedItem(item); err != nil {
+		if err := item.allocator.remover(item.id); err != nil {
```
Comment (Eabramov84) [inline spaceallocator/spaceallocator.go:544]: maybe remover should makes free space ?

Comment (al1img) [reply spaceallocator/spaceallocator.go:542]: It was initial idea, but to do not remove not necessary items, we decided to pass size on adding outdated items. As we know the size, spaceallocator can free required size on item remove. So, no additional actions in remove callback are required.

**Review by MykolaSolyanko — APPROVED** [review]

File: spaceallocator/spaceallocator.go (hunk 2 above)
Comment (al1img) [reply spaceallocator/spaceallocator.go]: Redundant changes. Initially, there were intermediate commits that I removed at final PR. Fixed.

**Review by Eabramov84 — APPROVED** [review]

**Pull request merged**